mobile/Yote/js/resources/product/views/SingleProduct.js

```diff
@@ -20,91 +20,78 @@ import {
-    super(props);
-    this._bind(
-      '_closeModal'
-      , '_openUpdateProduct'
-    )
-  }
+const SingleProduct = () => {
+  // get route state. Below is equivalent to getting navigation obj to access params
+  const route = useRoute();
+  // get navigation obj so we can access goBack(); 
+  const navigation = useNavigation(); 
 
-  componentDidMount() {
-    const { productId } = this.props.navigation.state.params;
-    this.props.dispatch(productActions.fetchSingleProductById(productId));
-  }
+  // get the product id from the route. Below is equivalent to const { productId } = this.props.navigation.state.params;
+  // this component knows about productId from declaring it in navigation stack in TabNavigator.js 
+  const { productId } = route.params; 
 
-  _closeModal() {
-    this.props.navigation.goBack();
-  }
+  // get the product from the store (or fetch it from the server)
+  const { data: product, ...productQuery } = useGetProductById(productId);
 
-  _openUpdateProduct() {
-    const { product } = this.props.navigation.state.params;
-    this.props.navigation.navigate('UpdateProduct', {product: product});
+  const leftItem = {
+    icon: require('../../../global/img/back.png'),
+    layout: 'icon',
+    onPress: () => navigation.goBack(),
   }
 
-  render() {
-    const { productMap } = this.props;
-    const { product } = this.props.navigation.state.params;
-    
-    const leftItem = {
-      icon: require('../../../global/img/back.png'),
-      layout: 'icon',
-      onPress: this._closeModal,
-    }
-
-    const rightItem = {
-      title: 'Edit'
-      , onPress: this._openUpdateProduct
-    }
-
-    return(
-      <View style={YTStyles.container}>
-        <YTHeader
-          title='Single Product'
-          leftItem={leftItem}
-          rightItem={rightItem}
-        />
-        <ScrollView>
+  return (
+    <View style={YTStyles.container}>
+      <YTHeader
+        title='Single Product'
+        leftItem={leftItem}
+      />
+      <WaitOn query={productQuery} fallback={<Skeleton />}>
+        <ScrollView style={{opacity: productQuery.isFetching ? 0.5 : null}}>
           <View style={{padding: 10}}>
-            <Text style={YTStyles.h1}>{product.title}</Text>
+            <Text style={YTStyles.h1}>{product?.title}</Text>
           </View>
           <View style={{paddingHorizontal: 10}}>
-            <Text style={YTStyles.text}>{product.description}</Text>
+            <Text style={YTStyles.text}>{product?.description}</Text>
+          </View>
+          <View style={{padding: 10}}>
+            <YTButton
+              caption={"Edit"}
+              onPress={() => navigation.navigate('UpdateProduct', {productId: product._id})}
+            />
           </View>
         </ScrollView>
-      </View>
-    )
-  }
+      </WaitOn>
+    </View>
+  )
 }
 
-SingleProduct.propTypes = {
-  productId: PropTypes.string
+const Skeleton = () => {
+  return (
+    <ListItem>
+      <Text>Loading</Text>
+    </ListItem>
+  )
```

Comment:
I think you'll want to use a different Skeleton here. ListItem doesn't seem like it belongs on SingleProduct view.